### PR TITLE
 Update APIM profile puppet resources for continuous update delivery

### PR DESCRIPTION
**Pull request by Savidude**

### CODE REVIEW

File: modules/apim_gateway/manifests/params.pp [hunk 1]
```diff
@@ -14,95 +14,32 @@
 #  limitations under the License.
 # ----------------------------------------------------------------------------
 
-# Claas apim::params
+# Class apim::params
```
Comment (VimukthiPerera) [inline modules/apim_gateway/manifests/params.pp]: Shouldn't this be apim_gateway params?

File: modules/apim_gateway/manifests/startserver.pp [hunk 2]
```diff
@@ -18,8 +18,14 @@
...
+  exec { 'daemon-reload':
+    command => "systemctl daemon-reload",
+    path    => "/bin/",
+  }
+
   service { $service_name:
-    ensure => running,
     enable => true,
+    ensure => running,
+    subscribe => File["binary"],
```
Comment (VimukthiPerera) [inline modules/apim_gateway/manifests/startserver.pp:29]: Format the code

File: modules/apim_gateway_master/build.sh [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+fi
+
+if [[ -s ${install_path}/bin/merged_files.txt ]]
+then
+  sed -i '1d' ${install_path}/bin/merged_files.txt # Remove first line from file
+  sed -i '$ d' ${install_path}/bin/merged_files.txt # Remove last line from file
+
+  while read -r line; do
+    filepath=${line##*${product}-${product_version}/}
+    template_file=${puppet_env}/modules/is_master/templates/carbon-home/${filepath}.erb
```
Comment (VimukthiPerera) [inline modules/apim_gateway_master/build.sh]: Shouldn't this be apim_gateway_master?

**Review by VimukthiPerera — CHANGES_REQUESTED** [review]

File: modules/apim_gateway_master/build.sh (hunk 3 above)
Comment (Savidude) [reply modules/apim_gateway_master/build.sh]: Yes! I'll fix that and all the other build scripts immediately.

File: modules/apim_gateway_master/manifests/startserver.pp [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+# ----------------------------------------------------------------------------
+
+# Class apim_gateway_master::startserver
+# Starts the server as a service in the final stage.
+class apim_gateway_master::startserver inherits apim_gateway_master::params {
+
+  # This class should be used to control Identity Server if required.
```
Comment (VimukthiPerera) [inline modules/apim_gateway_master/manifests/startserver.pp]: Not IS, should be APIM

File: modules/apim_km_master/manifests/startserver.pp [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+# ----------------------------------------------------------------------------
+
+# Class apim_km_master::startserver
+# Starts the server as a service in the final stage.
+class apim_km_master::startserver inherits apim_km_master::params {
+
+  # This class should be used to control Identity Server if required.
```
Comment (VimukthiPerera) [inline modules/apim_km_master/manifests/startserver.pp]: Should be API Manager here as well

Comment (VimukthiPerera) [inline modules/apim_km_master/manifests/startserver.pp]: Identity Server should be API Manager

File: modules/apim_store_master/manifests/startserver.pp [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+# ----------------------------------------------------------------------------
+
+# Class apim_store_master::startserver
+# Starts the server as a service in the final stage.
+class apim_store_master::startserver inherits apim_store_master::params {
+
+  # This class should be used to control Identity Server if required.
```
Comment (VimukthiPerera) [inline modules/apim_store_master/manifests/startserver.pp]: Please search and remove Identity server related comments

File: modules/apim_tm_master/manifests/startserver.pp [hunk 7]
```diff
@@ -0,0 +1,22 @@
+# ----------------------------------------------------------------------------
+#  Copyright (c) 2018 WSO2, Inc. http://www.wso2.org
```
Comment (VimukthiPerera) [inline modules/apim_tm_master/manifests/startserver.pp]: Copyright notices should be updated to 2019 as this is a new puppet module.

**Review by VimukthiPerera — APPROVED** [review]

File: modules/apim_gateway/manifests/params.pp (hunk 1 above)
Comment (Savidude) [reply modules/apim_gateway/manifests/params.pp]: Resolved https://github.com/wso2/puppet-apim/pull/86/commits/5dd25054bc378c8b256e4571b433b51eecc8d807

File: modules/apim_gateway/manifests/startserver.pp (hunk 2 above)
Comment (Savidude) [reply modules/apim_gateway/manifests/startserver.pp:29]: Resolved https://github.com/wso2/puppet-apim/pull/86/commits/5dd25054bc378c8b256e4571b433b51eecc8d807

File: modules/apim_gateway_master/build.sh (hunk 3 above)
Comment (Savidude) [reply modules/apim_gateway_master/build.sh]: Resolved https://github.com/wso2/puppet-apim/pull/86/commits/5dd25054bc378c8b256e4571b433b51eecc8d807

File: modules/apim_km_master/manifests/startserver.pp (hunk 5 above)
Comment (Savidude) [reply modules/apim_km_master/manifests/startserver.pp]: Resolved https://github.com/wso2/puppet-apim/pull/86/commits/5dd25054bc378c8b256e4571b433b51eecc8d807

File: modules/apim_gateway_master/manifests/startserver.pp (hunk 4 above)
Comment (Savidude) [reply modules/apim_gateway_master/manifests/startserver.pp]: Resolved https://github.com/wso2/puppet-apim/pull/86/commits/5dd25054bc378c8b256e4571b433b51eecc8d807

File: modules/apim_store_master/manifests/startserver.pp (hunk 6 above)
Comment (Savidude) [reply modules/apim_store_master/manifests/startserver.pp]: Resolved https://github.com/wso2/puppet-apim/pull/86/commits/5dd25054bc378c8b256e4571b433b51eecc8d807

File: modules/apim_tm_master/manifests/startserver.pp (hunk 7 above)
Comment (Savidude) [reply modules/apim_tm_master/manifests/startserver.pp]: Resolved https://github.com/wso2/puppet-apim/pull/86/commits/5dd25054bc378c8b256e4571b433b51eecc8d807

**Review by irunika — APPROVED** [review]

**Pull request merged**